tokio/tests/fs_canonicalize_dir.rs

```diff
@@ -0,0 +1,22 @@
+#![warn(rust_2018_idioms)]
+#![cfg(all(feature = "full", not(tokio_wasi)))] // WASI does not support all fs operations
```

Comment:
I noticed that this WASI comment that I copied was sometimes "file operations" and other times "directory operations". My apologies for possibly veering off-topic but from a little searching it looks to me like *some* fs operations may be possible with WASI but not all and WASI directory access would not work very easily with Tokio. I went ahead and used `sed` to apply a slightly more general statement across tokio/tests/fs_*.rs.